### ISSUE TITLE
Rename Metrics subsystems to align with naming convention

### ISSUE THREAD

**Issue by echappen** [18F/piipan#408]
Now that we have multiple function apps for metrics (for example, the api), we should rename the original function app that collects metrics from other systems to something more specific, as well as other components of the metrics system.

This will entail updating the ARM templates and other scripts that create these resources, and recreating them in the Azure dev environment. 

- Align with naming convention pattern, coordinate with @ryanhofdotgov

**Comment by echappen** [comment]
@ryanhofdotgov Was searching for our written comms about naming conventions and found this in our running meeting notes with FNS: 

> Use global FNS prefix (“FN”) on everything. Can also use “RG-FNS” for resource groups. That’s how USDA incident responders easily identify components as belonging to FNS


Is there documentation elsewhere I should also be referencing?

**Comment by echappen** [comment]
Also referencing https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/resource-naming

**Comment by ryanhofdotgov** [comment]
I'm tracking the larger renaming effort in https://github.com/18F/piipan/issues/252, which also links:
https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules

My [super early work in progress is here](https://docs.google.com/spreadsheets/d/1pYdvVoICjQjx9RvyYhuoV8D0xCrVRRQzYNhpdNvFsyc/edit#gid=1351188795), where I'm working out what resources need to be globally unique, and what have restrictive length limits/char restrictions.

I'm not a fan of the agency prefix the partner has asked to have on all resources; for our TTS sandbox I think we won't use them.

**Comment by echappen** [comment]
@ryanhofdotgov Here's what I'm thinking for Metrics resource names. How does this look to you?

| purpose | current name | final name |
| ------- | ------------ | ---------- |
| resource group | piipan-metrics | fns-rg-piipan-metrics-dev-westus-001 |
| secret keeper | metrics-secret-keeper | fns-kv-metrics-dev-westus-001 |
| database | piipan-metrics-db | fns-db-metrics-dev-westus-001 |
| Metrics API function app | PiipanMetricsApi[unique-string] | fns-func-metricsapi-dev-westus-001 |
| Metrics Collections function app | PiipanMetricsFunctions[unique-string] | fns-func-metricscollect-dev-westus-001 |
| Application Insights for API app | PiipanMetricsApi[unique-string] | fns-insights-metricsapi-dev-westus-001 |
| Application Insights Collections app | PiipanMetricsFunctions[unique-string] | fns-insights-metricscollect-dev-westus-001 |
| Storage for API app | metstor[unique-string] | fns-st-metricsapi-dev-westus-001 |
| Storage for Collections app | piipanmet[unique-string] | fns-st-metricscollect-dev-westus-001 |

Basically everything would be configurable on a name except the resource type (`rg`,`kv`, etc) and the app name (`metrics`, `metricsapi`, etc)

I checked the rules and I don't think I'll run into any length issues, but if I do, my plan is to remove from the end until I get to the environment indicator. If a name's still too long after that, then I'd need to tighten the app name or reconsider my approach. 

I also renamed `MetricsFunctions` to `MetricsCollections` to indicate that this is the app that collects the metrics.

**Comment by echappen** [comment]
> * The Azure Government cloud regions are named differently: Virginia/Texas/Arizona (perhaps with more to come in future?), so west/east nomenclature may not make sense here

^ Good to know, my plan is to make Azure location configurable

> * The partner side resource groups have already been named and created, at least in dev – the partner uses a slightly different naming convention for resource groups.

^ Ah, right

Let me know if you'd like me to add these new names to your spreadsheet.

**Comment by ryanhofdotgov** [comment]
@echappen The spreadsheet was just kind of a throwaway worksheet to just see upfront all the types of changes/restrictions I'd have to accommodate across the system. If you like, you can go ahead and create a PR for what you have done and I will tweak as necessary. For example: 
* I was thinking that we would only use the partner prefix in their environment
* I'll create external property files for TTS dev, TTS test (which hasn't been established yet), as well as partner-side dev.

The naming of resources isn't mandatory to get piipan deployed up in the partner environment, so it wasn't something I was necessarily tackling first.

**Comment by echappen** [comment]
PR for it is [here](https://github.com/18F/piipan/pull/451)

**Comment by ryanhofdotgov** [comment]
Even though we have an outstanding linked PR, I'm going to call this ticket done and address the remaining work in #252 and wrap that PR under the #252 scope.
